add feature comments

AlexMAK1 opened the pull request.

AlexMAK1 committed:
add feature comments

AlexMAK1 committed:
feature comments is done

NikolaiFufaev commented on main_service/src/main/java/ru/practicum/main_service/comments/CommentController.java:
```diff
@@ -0,0 +1,62 @@
+    private final CommentService commentService;
+
+    @PostMapping("/comments")
+    public CommentDto create(@Valid @RequestBody CommentDto commentDto) {
```
Не совсем понял логику, если вся информация должна придти в теле запроса, то явно не хватает проверок на null и наличие такого пользователя и эвента в бд. Если отправить пустое тело, то будет создан комментарий без привязок к событию и пользователю, и он сохранится в бд и получит id1. И точно ли любой пользователь(даже не зарегистрированный) может их оставлять?

AlexMAK1 replied:
Добавил проверки в слой сервиса во все методы. 
Изменил путь для эндпоинтов при создании, редактировании, удалении и т.д. 
Теперь эти возможности доступны только зарегистрированным пользователям.

NikolaiFufaev commented on main_service/src/main/resources/shema.sql:
```diff
@@ -0,0 +1,74 @@
+    CONSTRAINT pk_requests PRIMARY KEY (id),
+    FOREIGN KEY (event_id) REFERENCES events (event_id) ON DELETE CASCADE,
+    FOREIGN KEY (requester_id) REFERENCES users (user_id) ON DELETE CASCADE
+);
```
Не нашел в схеме таблицы с комментариями, хотя она не помешала бы.
Можно было бы настроить удаление комментариев каскадом при удалении пользователя или события, или хотя бы настроить связи на таблицы users и events

AlexMAK1 replied:
Добавил, таблицу в файл REDME.MD, при добавлении в схему возник конфликт с веткой develop, добавлю позже при мерже с main. Настроил удаление каскадом users и events.

NikolaiFufaev commented on main_service/src/main/java/ru/practicum/main_service/comments/model/Comment.java:
```diff
@@ -0,0 +1,23 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long commentId;
+    private String content;
+    private Boolean isPositive;
```
Какую роль играет isPositive?
Не понял зачем он нужен.

AlexMAK1 replied:
удалил, в данной реализации данное поле не нужно.

AlexMAK1 committed:
added validation in CommentServiceImpl,
changed path for endpoints in CommentController,
added table comments in README.md

AlexMAK1 committed:
deleted code for creation table comments for avoiding conflicts

AlexMAK1 committed:
the last changes
